src/dtj/java/com/lambdaherding/edi/dtj/ch08/InteractiveResponseBuilder.java

```diff
@@ -0,0 +1,42 @@
+package com.lambdaherding.edi.dtj.ch08;
+
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+public class InteractiveResponseBuilder{
+    private BiFunction<Consumer<String>, Supplier<String>, String> function;
```

Comment:
Arguably these could be consumers, instead of returning a String they are now capable of returning a response themselves. I'd like to make that change, it's the sort of semantic thing that gets left in as things evolve and you're not paying attention to the overall structure anymore